libraries/GCS_MAVLink/GCS_MAVLink_Parameters.cpp

```diff
@@ -0,0 +1,372 @@
+#define AP_MAV_DEFAULT_STREAM_RATE_POSITION 1
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA1 1
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA2 1
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA3 1
+#define AP_MAV_DEFAULT_STREAM_RATE_PARAMS 10
+#define AP_MAV_DEFAULT_STREAM_RATE_ADSB 5
+#elif APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_Blimp) || APM_BUILD_TYPE(APM_BUILD_Replay)
+#define AP_MAV_DEFAULT_STREAM_RATE_RAW_SENS 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXT_STAT 0
+#define AP_MAV_DEFAULT_STREAM_RATE_RC_CHAN 0
+#define AP_MAV_DEFAULT_STREAM_RATE_RAW_CTRL 0
+#define AP_MAV_DEFAULT_STREAM_RATE_POSITION 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA1 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA2 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA3 0
+#define AP_MAV_DEFAULT_STREAM_RATE_PARAMS 0
+#define AP_MAV_DEFAULT_STREAM_RATE_ADSB 0
+#elif APM_BUILD_TYPE(APM_BUILD_UNKNOWN)
+#define AP_MAV_DEFAULT_STREAM_RATE_RAW_SENS 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXT_STAT 0
+#define AP_MAV_DEFAULT_STREAM_RATE_RC_CHAN 0
+#define AP_MAV_DEFAULT_STREAM_RATE_RAW_CTRL 0
+#define AP_MAV_DEFAULT_STREAM_RATE_POSITION 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA1 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA2 0
+#define AP_MAV_DEFAULT_STREAM_RATE_EXTRA3 0
+#define AP_MAV_DEFAULT_STREAM_RATE_PARAMS 0
+#define AP_MAV_DEFAULT_STREAM_RATE_ADSB 0
+#else
+#error Need to set streamrates
+#endif  // APM_BUILD_TYPE
+#endif  // AP_MAV_DEFAULT_STREAM_RATE
+
+static const uint8_t default_rates[] {
+    AP_MAV_DEFAULT_STREAM_RATE_RAW_SENS,
+    AP_MAV_DEFAULT_STREAM_RATE_EXT_STAT,
+    AP_MAV_DEFAULT_STREAM_RATE_RC_CHAN,
+    AP_MAV_DEFAULT_STREAM_RATE_RAW_CTRL,
+    AP_MAV_DEFAULT_STREAM_RATE_POSITION,
+    AP_MAV_DEFAULT_STREAM_RATE_EXTRA1,
+    AP_MAV_DEFAULT_STREAM_RATE_EXTRA2,
+    AP_MAV_DEFAULT_STREAM_RATE_EXTRA3,
+    AP_MAV_DEFAULT_STREAM_RATE_PARAMS,
+    AP_MAV_DEFAULT_STREAM_RATE_ADSB,
+};
+
+#define DRATE(x) (float(default_rates[x]))
+
+/*
+  default stream rates to 1Hz
+ */
+const AP_Param::GroupInfo GCS_MAVLINK::var_info[] = {
+    // @Param: _RAW_SENS
+    // @DisplayName: Raw sensor stream rate
+    // @Description: MAVLink Stream rate of RAW_IMU, SCALED_IMU2, SCALED_IMU3, SCALED_PRESSURE, SCALED_PRESSURE2, SCALED_PRESSURE3 and AIRSPEED
+    // @Units: Hz
+    // @Range: 0 50
+    // @Increment: 1
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("_RAW_SENS", 1, GCS_MAVLINK, streamRates[0], DRATE(0)),
+
+    // @Param: _EXT_STAT
+    // @DisplayName: Extended status stream rate
+    // @Description: MAVLink Stream rate of SYS_STATUS, POWER_STATUS, MCU_STATUS, MEMINFO, CURRENT_WAYPOINT, GPS_RAW_INT, GPS_RTK (if available), GPS2_RAW_INT (if available), GPS2_RTK (if available), NAV_CONTROLLER_OUTPUT, FENCE_STATUS, and GLOBAL_TARGET_POS_INT
+    // @Units: Hz
+    // @Range: 0 50
+    // @Increment: 1
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("_EXT_STAT", 2, GCS_MAVLINK, streamRates[1], DRATE(1)),
```

Comment:
instead of '1', can we use STREAM_EXTENDED_STATUS ?

Reply:
Changed all of these.